lib/worker.js

```diff
@@ -274,24 +259,12 @@ class AssetComputeWorker {
 
         // post process and upload
         for (let rendition of this.renditions) {
-            // start post-processing timer - used for logging current interation
-            const postProcessingTimer = this.timers.currentPostProcessing = new Timer();
-            const shouldDoPostProcess = await this.shouldPostProcess(rendition);
-            if (shouldDoPostProcess) {
-                try {
-                    rendition = await this.doPostProcess(rendition);
-                    console.log(`Batch Postprocessing finished in ${postProcessingTimer.durationSec()} seconds: ${rendition.name}`);
-                    this.timers.postProcessing += postProcessingTimer.duration(); // add post processing rendition time to total post processing duration
-                } catch (err) {
-                    console.error(`rendition post-processing for ${rendition.name} failed with error after ${postProcessingTimer.durationSec()}: ${err.message || err}`);
-                    this.timers.postProcessing += postProcessingTimer.duration(); // add post processing rendition time to total post processing duration
-                    await this.renditionFailure(rendition, err);
-                    continue;
-                }
-            } else {
-                console.log(`No postprocessing done for rendition ${rendition.name}`);
-                this.timers.postProcessing += postProcessingTimer.duration(); // add post processing rendition time to total post processing duration
+
+            rendition = await this.postProcess(rendition);
```

Comment:
`postProcess` does not return something in all cases (if postprocessing is successful, there is no object/value returned. So the condition below will evaluate to true, always). At least I didn't see a return statement?

Reply:
It should and the tests forced me to get it right (spend some time on it). It returns undefined when there was a failure: https://github.com/adobe/asset-compute-sdk/blob/mob/NUI-668/lib/worker.js#L310